core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchReaderIterator.java

```diff
@@ -386,15 +386,12 @@ public void run() {
           fatalException = new TableDeletedException(tableId.canonical());
       } catch (SampleNotPresentException e) {
         fatalException = e;
-      } catch (Exception t) {
+      } catch (Throwable t) {
         if (queryThreadPool.isShutdown())
           log.debug("Caught exception, but queryThreadPool is shutdown", t);
         else
           log.warn("Caught exception, but queryThreadPool is not shutdown", t);
         fatalException = t;
-      } catch (Throwable t) {
-        fatalException = t;
-        throw t; // let uncaught exception handler deal with the Error
```

Comment:
I like the simplicity of this earlier solution. If the user is providing their own thread pools, why wouldn't they be able to provide their own uncaught exception handler to receive these and handle them on their own, if they wish?

Reply:
I resurrected the `throw t` line in 918b20f